app/post.go

```diff
@@ -798,20 +799,31 @@ func (a *App) parseAndFetchChannelIdByNameFromInFilter(channelName, userId, team
 }
 
 func (a *App) searchPostsInTeam(teamId string, userId string, paramsList []*model.SearchParams, modifierFun func(*model.SearchParams)) (*model.PostList, *model.AppError) {
-	channels := []store.StoreChannel{}
+	var wg sync.WaitGroup
+	wg.Add(len(paramsList))
+
+	//create a channel with buffer size of len(paramsList)
+	pchan := make(chan store.StoreResult, len(paramsList))
 
 	for _, params := range paramsList {
 		// Don't allow users to search for everything.
 		if params.Terms == "*" {
 			continue
 		}
 		modifierFun(params)
-		channels = append(channels, a.Srv.Store.Post().Search(teamId, userId, params))
+
+		go func(params *model.SearchParams) {
+			postList, err := a.Srv.Store.Post().Search(teamId, userId, params)
+			pchan <- store.StoreResult{Data: postList, Err: err}
```

Comment:
I think I missing here something like `wg.Done()` for each gorouting, actually if we are setting the WaitGroup to wait for the `len(paramsList)` we should add a `wg.Done()` in before the `continue` too.

Reply:
Still missing the `wg.Done()` before the `continue`

Reply:
nice catch. I removed the fixed length wg and instead adding to the wg inside the loop